aten/src/ATen/AccumulateType.h

```diff
@@ -22,6 +22,7 @@ struct AccumulateType { };
 
 #if defined(__CUDACC__) || defined(__HIPCC__)
 template <> struct AccumulateType<half, true> { using type = float; };
+template <> struct AccumulateType<BFloat16, true> {using type = float; };
```

Comment:
This is tricky. This is included in this PR since it is being used in `Distributions.cu`, for eg. https://github.com/ROCmSoftwarePlatform/pytorch/blob/a1ccdf83f6e0854166bd9118c3513d4b79a23c72/aten/src/ATen/native/cuda/Distributions.cu#L492.
And we are not making any distinction while dispatching kernels between cuda and rocm when using `AT_DISPATCH_FLOATING_TYPES_AND2`. It will fail compilation on cuda if we define this only for `__HIPCC__`.